runtime/doc/diagnostic.txt

```diff
@@ -671,9 +671,14 @@ Lua module: vim.diagnostic                                    *diagnostic-api*
 *vim.diagnostic.Opts.Status*
 
     Fields: ~
-      • {text}?  (`table<vim.diagnostic.Severity,string>`) A table mapping
-                 |diagnostic-severity| to the text to use for each severity
-                 section.
+      • {text}?    (`table<vim.diagnostic.Severity,string>`) A table mapping
+                   |diagnostic-severity| to the text to use for each severity
+                   section.
+      • {format}?  (`fun(sign:string,count:integer): string`) A function that
```

Comment:
Could we merge these 2 settings and allow `format` to be a function or a table of severity to string? That better aligns with other diagnostic config settings.